wrapanapi/systems/ec2.py

```diff
@@ -832,6 +832,35 @@ def list_templates(self, executable_by_me=True, owned_by_me=True, public=False):
         return [EC2Image(system=self, raw=self.ec2_resource.Image(image["ImageId"]))
                 for image in images]
 
+    def list_free_images(self, image_list=None):
+        """
+        Returns images which don't have a VM associated to it
+
+        Args:
+            image_list (list): List of  ids of all images in resource group
+        """
+        free_images = []
+        vm_list = self.list_vms()
+
+        if not vm_list:
```

Comment:
This block is unnecessary and repetitive.

```
for vm in vm_list:
    free_images.append()
return free_images 
```

This loop will noop when `vm_list` is an empty list, which the function is written to return.